src/lib/reasoners/sat_solver_sig.ml

```
@@ -75,6 +79,10 @@ module type S = sig
 
   (** [get_model t] produces the current model. *)
   val get_model : t -> unit
+
+  (** [get_unknown_reason t] returns the reason Alt-Ergo returned
+      [I_dont_know] if it did. If it did not, returns None. *)
+  val get_unknown_reason : t -> unknown_reason option
```

Maybe `raised` instead of `returned`?